core/src/main/java/com/amplifyframework/datastore/DataStoreCategoryBehavior.java

```diff
@@ -161,22 +161,40 @@
             @NonNull Consumer<DataStoreException> onQueryFailure
     );
 
-    /**
-     * Observe all changes to any/all item(s) in the DataStore.
-     * @param onObservationStarted Called when observation begins
-     * @param onDataStoreItemChange  Called 0..n times, once for each and every change that
-     *                               occurs to any/all item(s) in the DataStore.
-     * @param onObservationFailure   Called if observation of the DataStore terminates
-     *                               with a non-recoverable failure
-     * @param onObservationCompleted Called when observation completes gracefully
-     */
+
+        /**
+         * Observe all changes to any/all item(s) in the DataStore.
+         * @param onObservationStarted Called when observation begins
+         * @param onDataStoreItemChange  Called 0..n times, once for each and every change that
+         *                               occurs to any/all item(s) in the DataStore.
+         * @param onObservationFailure   Called if observation of the DataStore terminates
+         *                               with a non-recoverable failure
+         * @param onObservationCompleted Called when observation completes gracefully
+         */
     void observe(
             @NonNull Consumer<Cancelable> onObservationStarted,
             @NonNull Consumer<DataStoreItemChange<? extends Model>> onDataStoreItemChange,
             @NonNull Consumer<DataStoreException> onObservationFailure,
             @NonNull Action onObservationCompleted
     );
 
+    /**
+     * Query and Observe all changes to any/all item(s) in the DataStore.
+     * @param onObservationStarted Called when observation begins
+     * @param onQuerySnapshot  Called 0..n times, once for each and every change that
+     *                               occurs to any/all item(s) in the DataStore.
+     * @param onObservationError   Called if observation of the DataStore terminates
+     *                               with a non-recoverable failure
+     * @param onObservationComplete Called when observation completes gracefully
+     */
+    <T extends Model> void observeQuery(
+            @NonNull Class<T> itemClass,
+            @NonNull QueryOptions options,
+            @NonNull Consumer<Cancelable> onObservationStarted,
+            @NonNull Consumer<DataStoreQuerySnapshot<T>> onQuerySnapshot,
```

Comment:
It's not immediately clear how this is different than the existing `DataStoreItemChange<...>` -- can you update the docs? Is there a way to update the existing API to include the additional information being added here?

Reply:
-query returns an array of items
-observe returns a stream of items any time an item has changed.
-observeQuery returns a stream of arrays of items, any time the array changes (an item in the result set changed, or an item was added or removed to the result set)
 It is essentially a wrapper that combines the results of query and observe
We are working on the docs.